src/discof/replay/fd_replay_tile.c

```diff
@@ -1029,9 +1028,12 @@ send_tower_sync( fd_replay_tile_ctx_t * ctx ) {
   ulong vote_slot = fd_tower_votes_peek_tail_const( ctx->tower )->slot;
   fd_hash_t vote_bank_hash[1]  = { 0 };
   fd_hash_t vote_block_hash[1] = { 0 };
-  int err = fd_blockstore_bank_hash_query( ctx->blockstore, vote_slot, vote_bank_hash );
-  if( err ) FD_LOG_ERR(( "invariant violation: missing bank hash for tower vote" ));
-  err = fd_blockstore_block_hash_query( ctx->blockstore, vote_slot, vote_block_hash );
+
+  /* guaranteed to be on frontier from caller check */
+  fd_fork_t const * fork = fd_forks_query_const( ctx->forks, vote_slot );
+  fd_memcpy( vote_bank_hash, &fork->slot_ctx->slot_bank.banks_hash, sizeof(fd_hash_t) );
+
+  int err = fd_blockstore_block_hash_query( ctx->blockstore, vote_slot, vote_block_hash );
```

Comment:
How much does voting performance matter... ie, should we generate the vote as soon as we have the bank hash verses here?  ie, what is the cause/effect?  No change required here but lets think about changing the flow of operations...